### PR TITLE
GS-6 Task complete one off transaction dbus interface

### PR DESCRIPTION
**Pull request by miltonio94**
also separated files to make testing of library easier

### CODE REVIEW

**Comment by miltonio94** [comment]
I finished the dbus implementation for a one off transaction. A lot of hardcoded stuff for now, until we expand to other transaction types it will remain that way. I split the data types into their own modules in the library crate to make testing easier. Test are still pending for this, will do this later on in the week, possibly the weekend.

**Comment by miltonio94** [comment]
@murpjack maybe we should just merge this and come back to the tests later

File: src/lib/transaction.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    pub fn id(&self) -> i64 {
+        self.id
+    }
+    pub fn start_date(&self) -> i64 {
+        self.start_date
+    }
+}
+
+#[cfg(test)]
```
Comment (miltonio94) [inline src/lib/transaction.rs:72]: I think we could move this to a test folder so that we can have all our test in one place

**Comment by miltonio94** [comment]
I like what you did Jack, thanks for writing those tests and nice one on finding that uniqueness bug and fixing it

File: tests/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+use green_shark::transaction::{Currency, Frequency, Transaction};
```
Comment (miltonio94) [inline tests/transaction.rs:1]: Great, thanks for the change

**Review by murpjack — APPROVED** [review]

**Pull request merged**